### ISSUE TITLE
Bitwriter/reader

### ISSUE THREAD

**Issue by Jawsarn** [DarkRiftNetworking/DarkRift-Networking#45]
Adding a bitwriter reader and "finisher" that rounds up to nearest byte.

Basically 
WriteBits(uint, numBits) and a ReadBits(numBits) and then something like BitRoundup()

You can create some neat optimizations with tools like this for all kinds and purposes, and would be nice to be included in the API.

**Comment by JamJar00** [comment]
Sounds like a good idea, however I'm not clear what benefits something like this would have over a [BitArray](https://docs.microsoft.com/en-us/dotnet/api/system.collections.bitarray?view=netframework-4.7.2). Would this be for packing non-standard length types (like 7 bit ints) etc. or for something else?

I suppose BitArray could be a supported type quite easily anyway and that might be a good idea!

**Comment by Jawsarn** [comment]
Yes, for nonstandard types. And I guess reduce middle code in cases where you don't know the size of the bit-array beforehand as well.^^

**Comment by JamJar00** [comment]
Ok, think I can sort something like that at some point!
